shinylive/_url.py

```diff
@@ -0,0 +1,96 @@
+import base64
+import json
+from pathlib import Path
+from typing import List, Literal, Optional, TypedDict, cast
+
+from lzstring import LZString  # type: ignore[reportMissingTypeStubs]
+
+
+def make_shinylive_url(
+    app: str | Path,
```

Comment:
Here `app` is a path to a file on disk, but I can see utility in allowing `app` to be the actual code of a single `app.py` or `app.R` file. That doesn't need to make it into this PR but I'd be happy for advice about how to design the API in a pythonic way.